diff --git a/SConstruct b/SConstruct
index a6f792c..038cd66 100644
--- a/SConstruct
+++ b/SConstruct
@@ -147,6 +147,9 @@ opts.Add(BoolVariable("generate_template_get_node", "Generate a template version
 opts.Update(env)
 Help(opts.GenerateHelpText(env))
 
+if env["platform"] != "windows":
+    env.Append(CXXFLAGS=["-std=c++17"])
+
 # This makes sure to keep the session environment variables on Windows.
 # This way, you can run SCons in a Visual Studio 2017 prompt and it will find
 # all the required tools
@@ -165,7 +168,7 @@ if env["platform"] == "linux" or env["platform"] == "freebsd":
     if env["use_llvm"]:
         env["CXX"] = "clang++"
 
-    env.Append(CCFLAGS=["-fPIC", "-std=c++17", "-Wwrite-strings"])
+    env.Append(CCFLAGS=["-fPIC", "-Wwrite-strings"])
     env.Append(LINKFLAGS=["-Wl,-R,'$$ORIGIN'"])
 
     if env["target"] == "debug":
@@ -187,7 +190,7 @@ elif env["platform"] == "osx":
     if env["bits"] == "32":
         raise ValueError("Only 64-bit builds are supported for the macOS target.")
 
-    env.Append(CCFLAGS=["-std=c++17", "-arch", env["macos_arch"]])
+    env.Append(CCFLAGS=["-arch", env["macos_arch"]])
 
     if env["macos_deployment_target"] != "default":
         env.Append(CCFLAGS=["-mmacosx-version-min=" + env["macos_deployment_target"]])
@@ -234,7 +237,7 @@ elif env["platform"] == "ios":
     env["AR"] = compiler_path + "ar"
     env["RANLIB"] = compiler_path + "ranlib"
 
-    env.Append(CCFLAGS=["-std=c++17", "-arch", env["ios_arch"], "-isysroot", sdk_path])
+    env.Append(CCFLAGS=["-arch", env["ios_arch"], "-isysroot", sdk_path])
     env.Append(
         LINKFLAGS=[
             "-arch",
@@ -258,6 +261,7 @@ elif env["platform"] == "windows":
         # MSVC
         env.Append(CPPDEFINES=["TYPED_METHOD_BIND"])
         env.Append(LINKFLAGS=["/WX"])
+        env.Append(CCFLAGS=["/std:c++17"])
         if env["target"] == "debug":
             env.Append(CCFLAGS=["/Z7", "/Od", "/EHsc", "/D_DEBUG", "/MDd"])
         elif env["target"] == "release":
